lightbeam/api.py

```diff
@@ -50,32 +50,63 @@ def prepare(self, selector="*"):
         all_endpoints = self.get_sorted_endpoints()
 
         # filter down to only selected endpoints
+        self.lightbeam.endpoints = self.apply_filters(all_endpoints)
+
+
+    def apply_filters(self, endpoints=[]):
```

Comment:
I've rewritten the apply_filters function to be easier to read. Please take a look and test this alternative and let me know if it works:

```
def apply_filters(self, endpoints: List[str] = list()):
    selected_endpoints = self.parse_endpoint_string(self.lightbeam.selector, endpoints=endpoints, all_on_empty=True)

    # make sure all selectors resolve to an endpoint
    unknown_endpoints = list(set(selected_endpoints).difference(endpoints))
    if unknown_endpoints:
        self.logger.critical("no match for selector(s) [{0}] to any endpoint in your API; check for typos?".format(", ".join(unknown_endpoints)))

    excluded_endpoints = self.parse_endpoint_string(self.lightbeam.exclude, endpoints=selected_endpoints)
    
    # make sure we have some endpoints to process
    my_endpoints = list(set(selected_endpoints).difference(excluded_endpoints))
    if not my_endpoints:
        self.logger.critical("selector filtering left no endpoints to process; check your selector for typos?")

    return my_endpoints


@staticmethod
def parse_endpoint_string(full_endpoint_string: str, endpoints: List[str], all_on_empty=False) -> List[str]:
    """
    Possible endpoint strings:
    - "students"
    - "students,schools"
    - "student*"
    - "student*,schools"
    - "*Associations"
    - "*Associations,schools"
    """
    # If no string is provided, return all or no endpoints, depending on use-case.
    if not full_endpoint_string:
        if all_on_empty:
            return endpoints
        else:
            return []
    
    # Asterisk wildcards to all endpoints.
    if full_endpoint_string == "*":
        return endpoints
    
    # Otherwise, a comma-separated list of endpoints is expected.
    return_endpoints = set()

    for endpoint_string in full_endpoint_string.split(","):

        if endpoint_string.startswith("*"):  # left wildcard: "*Associations"
            return_endpoints.update(
                filter(lambda endpoint: endpoint.endswith(endpoint_string.lstrip("*")), endpoints)
            )
        elif endpoint_string.endswith("*"):  # right wildcard: "student*"
            return_endpoints.update(
                filter(lambda endpoint: endpoint.startswith(endpoint_string.rstrip("*")), endpoints)
            )
        else:  # no wildcard: "students"
            return_endpoints.add(endpoint_string)
    
    return list(return_endpoints)
```

Reply:
I tested this and it works (though I did remove some of the typing so I wouldn't have to import List, etc.). Feel free to argue against this if you feel strongly, my reasoning was mainly that I don't have argument typing throughout the rest of `lightbeam` so I removed for consistently. Maybe in the future we can do a refactor where we add typing throughout.